Sources/Penny/CommandsManager.swift

```diff
@@ -91,46 +91,41 @@ enum SlashCommand: String, CaseIterable {
         switch self {
         case .howManyCoinsApp:
             return .user
-        case .link, .autoPings, .faqs, .howManyCoins:
+        case .github, .autoPings, .faqs, .howManyCoins:
             return nil
         }
     }
 }
 
-enum LinkSubCommand: String, CaseIterable {
-    case discord, github, slack
+enum GitHubSubCommand: String, CaseIterable {
+    case help
+    case link
+    case unlink
 
     var description: String {
         switch self {
-        case .discord:
-            return "Link your Discord account"
-        case .github:
-            return "Link your GitHub account"
-        case .slack:
-            return "Link your Slack account"
+        case .help:
+            return "Help about how GitHub linking works"
+        case .link:
+            return "Link your GitHub account to Penny"
+        case .unlink:
+            return "Unlink your GitHub account from Penny"
         }
     }
 
     var options: [ApplicationCommand.Option] {
         switch self {
-        case .discord: return [.init(
-            type: .string,
-            name: "id",
-            description: "Your Discord account ID",
-            required: true
-        )]
-        case .github: return [.init(
-            type: .string,
-            name: "id",
-            description: "Your GitHub account ID",
-            required: true
-        )]
-        case .slack: return [.init(
-            type: .string,
-            name: "id",
-            description: "Your Slack account ID",
-            required: true
-        )]
+        case .help:
+            return []
+        case .link:
+            return [.init(
```

Comment:
Now that we discussed everything on Discord, we know that we don't need the username option at all.

What happens is that Discord sends an `Interaction` payload our way, and the payload contains the Discord info of the user, including the Discord ID. so we can do something like `interaction.author.id` and get their ID.

The Interaction also contains a unique token, with which we can edit the interaction again from the Lambda.
So what happens is that we respond to the interaction in Discord part of penny and say "hey click on this link to link/button your GH". The link contains points to GitHub oauth-authorize endpoint, and the link contains an `state` parameter. 
The state parameter is the JWT token which you're supposed to create and sign using the **private** key. The `GHOAuthPayload` needs a `DiscordID` and `InteractionToken` which i mentioned both.

On the other side, the lambda gets the state parameter, verifies the state as you've done, and it'll have access to the same `GHOAuthPayload`.
After we have the OAuth access_token, the lambda uses the `discordID` in the `GHOAuthPayload` to do a penny-dynamo-db-table query and find the discord user, then update their GitHub ID and save the user to dynamo db. And linking is done :)